### ISSUE TITLE
Support Value metadata (and in particular comments)

### ISSUE THREAD

**Issue by jomala** [serde-rs/serde#2095]
I can see a number of applications for being able to attach metadata to deserialised Values. Primarily, it would be useful to be able to deserialise in one format and reserialise in another while preserving data which doesn't fit the serde model.

* An example might be preserving comments during YAML to TOML conversion. I realise the standards don't define how comments should be associated with particular fields in YAML, but there are common conventions that could be encoded. And that few parsing libraries (except tomllib) support comments currently.
* Another would be some form of preservation of formatting and comments after data manipulation (e.g. https://github.com/dtolnay/serde-yaml/issues/100).
* This would also allow programmatic addition of comments to output (e.g. https://github.com/dtolnay/serde-yaml/issues/145)

I assume this wouldn't be a feature that the serde team would be interested in developing, but if it was developed (behind a feature flag), might it be of interest as a pull request?

**Comment by dtolnay** [comment]
I don't think this would be a good fit for serde. You would need to use a different library or some kind of postprocessor to insert the comments.
